pandas/core/internals/managers.py

```diff
@@ -1947,9 +1947,17 @@ def _blklocs(self):
         """compat with BlockManager"""
         return None
 
-    def getitem_mgr(self, indexer: slice | npt.NDArray[np.bool_]) -> SingleBlockManager:
+    def getitem_mgr(self, indexer: slice | np.ndarray) -> SingleBlockManager:
         # similar to get_slice, but not restricted to slice indexer
         blk = self._block
+        if (
+            _using_copy_on_write()
+            and isinstance(indexer, np.ndarray)
+            and len(indexer) > 0
+            and com.is_bool_indexer(indexer)
+            and indexer.all()
+        ):
+            return type(self)(blk, self.index, [weakref.ref(blk)], parent=self)
```

Comment:
Is this special case needed here? 
For example for `dropna` itself, this check is already done on a level higher up. And so doing that here again might mean the check is done multiple times?

Reply:
Ah, I see we only do the check higher up in DataFrame.dropna, while this one is for Series.dropna.

(I am still not fully sure if I find it worth the added complexity though)

Reply:
I'd add It for now, can remove if we don't think it's worth the complexity in the long run?